tools/testjson2md/testjson2md.go

```diff
@@ -95,14 +102,21 @@ func markdownForContent(content report.DisplayContent) ([]byte, error) {
 		len(content.Results["pass"]), len(content.Results["fail"]), len(content.Results["skip"]))
 
 	// failed tests
+	var totalSize int
 	if len(content.Results["fail"]) > 0 {
 		fmt.Fprintf(&buf, "#### Failed Tests\n")
 		for _, test := range content.Results["fail"] {
-			fmt.Fprintf(&buf, "<details><summary>%s</summary>\n\n", test.TestName)
-			fmt.Fprintf(&buf, "```code\n")
-			fmt.Fprintf(&buf, "%s", testEventToString(lastN(test.Events, *truncateLines)))
-			fmt.Fprintf(&buf, "```\n")
-			fmt.Fprintf(&buf, "</details>\n")
+			s, d := testEventToDetailsBlock(test.TestName, test.Events)
+			// check if we are over the limit
+			if totalSize+s > (summaryLimitInBytes - summaryReservedInBytes) {
```

Comment:
What if you simply leave the Failed Tests by last (moving the `Test Durations` section to the beginning maybe)?

Reply:
It drops the entire summary with error annotation: `$GITHUB_STEP_SUMMARY upload aborted, supports content up to a size of 1024k, got 1265k.`.  We can definitely truncate returning slice but that might result in broken markdown. Since it one off case I am fine with that. Let me know if you think that will be better?